src/ray/util/signal_handler.cc

```diff
@@ -0,0 +1,85 @@
+#include "ray/util/signal_handler.h"
+
+#include <limits.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sstream>
+
+#include "ray/util/logging.h"
+
+using namespace ray;
+
+// Normally stop ray will also send the signal.
+int SignalHandlers::terminate_logging_level_ = RAY_INFO;
+// The list of signals that has installed handlers.
+std::vector<int> SignalHandlers::installed_signals_;
+// The current app name.
+std::string SignalHandlers::app_name_;
+
+void InstallSignalHandlerHelper(int signal, const struct sigaction &action,
+                                std::vector<int> &installed_signals) {
```

Comment:
According to [Google code style](https://google.github.io/styleguide/cppguide.html#Output_Parameters), it will be better to make the input/output parameter `installed_signals` be a pointer to `std::vector<int>`.

But it's not a hard-and-fast rule.